app/controllers/application_controller.rb

```diff
@@ -1,20 +1,46 @@
-# typed: true
+# typed: false
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  extend T::Sig
+
   include DeviseRemoteUser::ControllerBehavior
 
   # Smartly redirect user back to URL they requested before authenticating
   # From: https://github.com/heartcombo/devise/wiki/How-To:-Redirect-back-to-current-page-after-sign-in,-sign-out,-sign-up,-update
   before_action :store_user_location!, if: :storable_location?
+  # before_action :authenticate_user!
+
+  rescue_from ActionPolicy::Unauthorized, with: :deny_access
+
+  sig { returns(T.nilable(User)) }
+  def current_user
+    super.tap do |cur_user|
+      break unless cur_user
+
+      cur_user.groups = ldap_groups
+    end
+  end
```

Comment:
implementation based on patterns used in hydrus and argo, per ticket suggestion

Reply:
also, per jcoyne's suggestion, i played for a couple hours with having a `UserWithGroups` class, that has an instance of the `User` active record class, but i couldn't find a way to get devise to play nicely with a  setup where the object returned by `current_user` isn't of the class that's used for database persistence of the user record.  and my takeaway from that conversation with @jcoyne was that that was the point of trying that, since having `groups` and `groups=` methods on an AR object could be confusing, since we don't persist that info (we always want it from the request env headers).